addons/account/tests/test_account_move_in_invoice.py

```diff
@@ -2025,3 +2025,55 @@ def test_bill_amount_should_be_editable(self):
                                     'debit': 500,
                                     'tax_line_id': self.company_data['default_tax_purchase'].id,
                                 }])
+
+    def test_invoice_sent_to_additional_partner(self):
+        """
+        Make sure that when an invoice is a partner to a partner who is not
+        the invoiced customer, they receive a link containing an access token,
+        allowing them to view the invoice without needing to log in.
+        """
+
+        # Create a simple invoice for the partner
+        invoice = self.env["account.move"].create({
+            "partner_id": self.partner_a.id,
+            "move_type": "out_invoice",
+            "invoice_date": "2023-04-17",
+            "invoice_line_ids": [(0, 0, {
+                "name": "Test Invoice Line",
+                "quantity": 1,
+                "price_unit": 100,
+            })],
+        })
+
+        # Set the invoice to the 'posted' state
+        invoice.action_post()
+
+        # Create a partner not related to the invoice
+        additional_partner = self.env["res.partner"].create({
+            "name": "Additional Partner",
+            "email": "additional@example.com",
+        })
+
+        # Send the invoice
+        action = invoice.action_invoice_sent()
+        action_context = action["context"]
+
+        # Create the email using the wizard and add the additional partner as a recipient
+        invoice_send_wizard = self.env["account.invoice.send"].with_context(
+            action_context,
+            active_ids=[invoice.id]
+        ).create({'is_print': False})
+        invoice_send_wizard.partner_ids |= additional_partner
+
+        invoice_send_wizard.template_id.auto_delete = False
```

Comment:
Not sure what this step is

Reply:
`invoice_send_wizard.partner_ids |= additional_partner` is for adding a recipient to the mail.

`invoice_send_wizard.template_id.auto_delete = False`:
When `auto_delete` is `true`, the associated `mail.mail` record is automatically deleted after being sent.
Here, we disable this option because we need that record for testing.

Reply:
I'm dumb, I should have asked you to add a comment explaining that in the code. :/
Too bad.